docs/data/migration/migration-pickers-v7/migration-pickers-v7.md

```diff
@@ -312,6 +312,59 @@ const theme = createTheme({
 });
 ```
 
+### Clean the `ownerState`
+
+The `ownerState` is an object describing the current state of a given component to let you do more advanced customization.
+It is used in two different APIs:
+
+1. In the theme's `styleOverrides`, the `ownerState` is passed to the function that returns the style:
+
+```ts
+const theme = createTheme({
+  components: {
+    MuiDateCalendar: {
+      styleOverrides: {
+        root: ({ ownerState }) => ({
+          /** Style based on the ownerState */
+        }),
+      },
+    },
+  },
+});
+```
+
+2. In the `slotProps`, the `ownerState` is passed to the function that returns the custom props:
+
+```tsx
+<DatePicker
+  slotProps={{
+    actionBar: (ownerState) => ({
+      /** Props based on the ownerState */
+    }),
+  }}
+/>
+```
+
+Before version `v8.x`, the `ownerState` was equal to the props of the component that was defining it, plus some additional internal states if needed.
+This came with a few problems:
+
+- It is hard to know which parent defines the `ownerState` and therefore which props it contains (is the `actionBar` slot handled by `DatePicker`? by `DesktopDatePicker`? by `PickerLayout`?).
+
+- A lot of the properties of the `ownerState` were not meant for public usage, which made the evolution of the codebase complex because all the props received by an internal component becomes public API and changing them becomes a breaking change.
```

Comment:
```suggestion
- Many properties of the `ownerState` were not meant for public usage, which made the evolution of the codebase complex because all the props received by an internal component became public API and any change introduced a breaking change.
```